app/models/searchgov_url.rb

```diff
@@ -15,15 +15,26 @@ class SearchgovUrl < ActiveRecord::Base
 
   attr_accessible :last_crawl_status, :last_crawled_at, :url
   attr_reader :response, :document
+  attr_readonly :url
+
+  validates_associated :searchgov_domain, on: :create
+  validates_presence_of :searchgov_domain, on: :create
 
   validates :url, uniqueness: true
   validates :url_extension,
     exclusion: { in: BLACKLISTED_EXTENSIONS,  message: "is not one we index" },
     allow_blank: true
 
   before_validation :escape_url
+  before_validation :set_searchgov_domain, on: :create
   before_destroy :delete_document
 
+  belongs_to :searchgov_domain
+  counter_culture :searchgov_domain, column_name: 'urls_count'
```

Comment:
I read up on the `counter_culture` gem since I had never heard of it. I think I get what's going on here. This is what's responsible for keeping `searchgov_domains.urls_count` and `searchgov_domains.unfetched_urls_count` up to date 👍.